### DISCUSSION TITLE
Implementing the custom timeframe

### DISCUSSION THREAD

**Discussion by tehicee** [mivano/azure-cost-cli#29] [Q&A] [answered]
I've implemented the following command, 
azure-cost -s ***** -t --timeframe --from 2023-01-01 --to 2023-04-31

The error message received is in the screenshot included.

Thank you!

I've found the solution based on the previous discussion,
1) Update a new version of the NuGet
2) azure-cost -s xxxx -t Custom --from 2023-01-01 --to 2023-04-30

![Screenshot 2023-05-09 145442](https://user-images.githubusercontent.com/118544209/237017868-878dc74c-1dfd-4fc1-aa39-296c7ae56a46.png)

**Comment by mivano** [comment, marked as answer]
I will make sure the documentation is a bit more clear on this. Thanks for reporting.

**Comment by niklasfink** [comment]
@mivano are there other predefined timeframes which can be used? Such as LastMonth or LastQuater or similar.

**Reply by mivano** [reply]
I said I would update the docs, but it looks like I forgot to do so. I just added a section for it: https://github.com/mivano/azure-cost-cli#timeframe
Hope this helps.

**Reply by jashjain21** [reply]
Hey @mivano, could you provide an example of using the "TheLastMonth" attribute in the timeframe?
This is what I tried and it failed
`azure-cost accumulatedCost -s "sss" -g "rrr" -t TheLastMonth`

**Reply by mivano** [reply]
Interesting, the error message is `Invalid query definition, timeframe TheLastMonth is currently not supported.`
Not sure how long that _currently_ is though. Thanks for reporting, I will need to dive into it. Try to use `-t Custom --from --to` instead.
